libs/wingsdk/src/cloud/website.ts

```diff
@@ -91,7 +91,27 @@ export abstract class Website extends Resource {
    * @param path the file path it will be uploaded as
    * @param data the data to write to the file
    */
-  public abstract addJson(path: string, data: Json): string;
+  public addJson(path: string, data: Json): string {
+    if (!path.endsWith(".json")) {
+      throw new Error(
+        `key must have a .json suffix. (current: "${path.split(".").pop()}")`
+      );
+    }
+    return this.addFile(path, JSON.stringify(data), "application/json");
+  }
+
+  /**
+   * Add a file with custom value during the website's deployment.
+   * If the path conflicts with file path from the website's static assets, an error will be thrown.
+   * @param path the file path it will be uploaded as
+   * @param data the data to write to the file
+   * @param contentType the file content-type
+   */
+  public abstract addFile(
+    path: string,
+    data: string,
+    contentType: string
+  ): string;
```

Comment:
I think this method name could confuse users since the methods `addFile` exists on `Bucket` and `Website` but they have different meanings:

- In `Bucket`, `addFile()` expects a file path (typed as `str`)
- In `Website`, `addFile()` expects raw data (typed as `str`)

Maybe we can rename the second one to `addFileData` for now? (I don't love this name - open to suggestions)

Reply:
maybe `createFileFromStr` (on the website)?

Reply:
I'll postpone it to a following PR